lib/identity_cache/query_api.rb

```diff
@@ -179,9 +175,43 @@ def require_if_necessary #:nodoc:
       end
 
       def resolve_cache_miss(id)
-        object = self.includes(cache_fetch_includes).reorder(nil).where(primary_key => id).first
-        object.send(:populate_association_caches) if object
-        object
+        record = self.includes(cache_fetch_includes).reorder(nil).where(primary_key => id).first
+        preload_id_embedded_associations([record]) if record
+        record
+      end
+
+      def preload_id_embedded_associations(records)
+        return if records.empty?
+        each_id_embedded_association do |name, options|
+          reflection = options.fetch(:association_reflection)
+          child_model = reflection.klass
+          scope = child_model.all
+          scope = scope.instance_exec(nil, &reflection.scope) if reflection.scope
+
+          pairs = scope.where(reflection.foreign_key => records.map(&:id)).pluck(reflection.foreign_key, reflection.active_record_primary_key)
+          ids_by_parent = {}
+          pairs.each do |parent_id, child_id|
+            (ids_by_parent[parent_id] ||= []) << child_id
```

Comment:
Maybe defining the default value to hash with `[]` would make this method simpler?